tests/compilation/test_compile.py

```diff
@@ -141,3 +141,48 @@ def test_compile_errors(routine, expected_error, backend):
         compile_routine(
             routine, preprocessing_stages=[introduce_port_variables], backend=backend, skip_verification=True
         )
+
+
+def test_compilation_introduces_constraints_stemming_from_relation_between_port_sizes(backend):
```

Comment:
I think it's also worth adding a `fun(N)` to make sure parsing functions also works in this context?

Reply:
Done!